frontend/src/app/modules/bim/bcf/bcf-wp-single-view/bcf-wp-single-view.component.ts

```diff
@@ -2,49 +2,76 @@ import {Component, Input, OnDestroy, OnInit} from "@angular/core";
 import {StateService} from "@uirouter/core";
 import {WorkPackageResource} from "core-app/modules/hal/resources/work-package-resource";
 import {NgxGalleryImage, NgxGalleryOptions} from '@kolkov/ngx-gallery';
+import {PathHelperService} from "core-app/modules/common/path-helper/path-helper.service";
 import {HalLink} from "core-app/modules/hal/hal-link/hal-link";
+import {BcfApiService} from "core-app/modules/bim/bcf/api/bcf-api.service";
+import {BcfViewpointPaths} from "core-app/modules/bim/bcf/api/viewpoints/bcf-viewpoint.paths";
+import {CurrentProjectService} from "core-components/projects/current-project.service";
+import {ModelViewerService} from "core-app/modules/bcf/services/model-viewer.service";
+import {I18nService} from "core-app/modules/common/i18n/i18n.service";
+
+export type ViewPointOriginal = { uuid:string, snapshot_id:string, snapshot_file_name:string };
+export type ViewPoint = { snapshotId:string, snapshotFileName?:string, snapshotFullPath:string };
 
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><del><span data-href="https://github.com/opf/openproject/blob/fcd8020426c30218b3da5ed1c5f3e559963ef495/frontend/src/app/modules/bim/bcf/bcf-wp-single-view/bcf-wp-single-view.component.ts#L15"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</del></td>
    </tr>
  </tbody>
</table>